src/tests/tests/background.js

```diff
@@ -357,4 +358,53 @@ QUnit.test("subdomains on the yellowlist are preserved", (assert) => {
   );
 });
 
+QUnit.test("migrates misattributed MDFP domains from snitch map", (assert) => {
+  setupBadgerStorage(badger);
+
+  // store one action map for all tests (should always be preserved)
+  const actionMap = badger.storage.getBadgerStorageObject('action_map'),
+    snitchMap = badger.storage.getBadgerStorageObject('snitch_map');
+
+  // reset snitchMap's storage to blank object
+  snitchMap.updateObject({});
+
+  // snitch map with no MDFP entries
+  let snitchNoMDFP = {
+    ['amazon.com']: ['amazonads.com', 'amazing.com', 'amazonrainforest.com']
+  };
+
+  // snitch map with some MDFP entries
+  let snitchSomeMDFP = {
+    ['amazon.com']: ['amazon.ca', 'amazon.co.jp', 'amazing.com']
+  };
+
+  // snitch map with all MDFP entries
+  let snitchAllMDFP = {
+    ['amazon.com']: ['amazon.ca', 'amazon.co.jp', 'amazon.es']
+  };
+
+  // tests that a snitchMap entry with no MDFP domains stays the same after the migration
+  snitchMap.updateObject(snitchNoMDFP);
+  migrations.forgetFirstPartySnitches(badger);
+  assert.ok(badger.storage.getBadgerStorageObject('snitch_map').getItem('amazon.com'),
+    'forget first party migration does not affect a snitchmap with no misattributed MDFP items'
+  );
+
+  // tests that a snitchMap entry with some MDFP domains has those items removed but remains in the snitch Map
+  snitchMap.updateObject(snitchSomeMDFP);
+  migrations.forgetFirstPartySnitches(badger);
+  assert.equal(snitchMap.getItem('amazon.com')[0] === 'amazing.com' && snitchMap.getItem('amazon.com').length === 1, true,
+    'forget first party migration properly removes MDFP domains and leaves regular domains');
+
+  // tests that a snitchMap entry with all MDFP domains will have all domains removed then removed entirely from snitchMap
+  snitchMap.updateObject(snitchAllMDFP);
+  migrations.forgetFirstPartySnitches(badger);
+  assert.notOk(snitchMap.getItem('amazon.com'),
+    'forget first party migration properly removes a snitch map entry with all MDFP domains attributed to it');
+
+
+  // action map stays the same throughout all tests
+  assert.equal(actionMap, badger.storage.getBadgerStorageObject('action_map'), 'action map stays the same after running migration');
```

Comment:
Should be `assert.deepEqual()`, and should be comparing `getItemClones()`. But I don't think this check makes sense: action map should indeed change for every test. We should set up and verify the contents of action map for each of the three scenarios.

Reply:
Something like:
```javascript
  let actionNoMDFP = {
    'amazon.com': {
      heuristicAction: "cookieblock",
      userAction: "",
      dnt: false,
      nextUpdateTime: 0,
    }
...
  snitchMap.updateObject(snitchNoMDFP);
  actionMap.updateObject(actionNoMDFP);
  migrations.forgetFirstPartySnitches(badger);
...
  assert.deepEqual(
    actionMap.getItem('amazon.com'),
    actionNoMDFP['amazon.com'],
    "action map preserved for domain with no MDFP snitch entries"
  );
...
  assert.equal(
    badger.storage.getAction('amazon.com'),
    constants.ALLOW,
    "Action downgraded for partial MDFP domain"
  );
...
```